### ISSUE TITLE
IWYU requires you to include header files for enums but there's no way you can't have included it 

### ISSUE THREAD

**Issue by ThosRTanner** [include-what-you-use/include-what-you-use#244]
For instance, with QT, you have

    QFile::open(QIODevice::some flag)

Because this parameter is an enum and enums can't be forward declared, then the include has to be there for QFile::open to be declared.

So when in your code you then do

    QFile file("xxx");
    file.open(QIODevice::Read)

include what you use says you are missing `#include <QIODevice>`.

But it has to be implicitly available because of the way open is declared.

**Comment by vsapsai** [comment]
Good point. IWYU discourages transitive includes but it looks reasonable to treat enum as part of exported API. Still need to see how it integrates with existing re-export cases.

Note: this is applicable to traditional enums, strongly typed enums is a different beast.

**Comment by ThosRTanner** [comment]
Oh, quite agree about strongly typed ones (on the basis you can forward declare them anyway).

**Comment by firewave** [comment]
Suggesting enum forward declaration was added a few versions ago.
